src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -806,11 +806,14 @@ private void BeforeQueryStatusForPowerConsole(object sender, EventArgs args)
                 NuGetUIThreadHelper.JoinableTaskFactory.Run(InitializeMEFAsync);
             }
 
+            var isConsoleBusy = false;
             if (ConsoleStatus != null)
             {
-                var command = (OleMenuCommand)sender;
-                command.Enabled = !ConsoleStatus.Value.IsBusy && !_powerConsoleCommandExecuting;
+                isConsoleBusy = ConsoleStatus.Value.IsBusy;
```

Comment:
You could dedupe a lot of code you're adding in this PR with a helper property (`IsConsoleBusy`) or helper method (`GetIsConsoleBusy()`).